### PR TITLE
Nessie: Prevent accidental deletion of files which are still referenced by other branches/tags

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Iceberg GC functionalities like `remove_orphan_files`, `expire_snapshots`, `drop_table` (with `purge=true`) are not aware of the Nessie's branches and tags. 
So, when a user accidentally calls these on Nessie-managed tables, these GC functionalities will clean up the files which are still referenced in other branches/tags. So, blocking all the Iceberg GC operations on Nessie-managed tables by default. 

Nessie will provide engine agnostic, reference-aware GC functionalities (which doesn't call these Iceberg GC functionalities internally) with CLI support for handling the GC of expired/unreferenced files of a Nessie managed table.

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Review by dimas-b — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
Can one of the committers please help merge this PR? 
Also please include this in `1.0 milestone` or immediate next release milestone (would like to have this fix from the next version itself)

cc: @rdblue, @pvary, @Fokko, @RussellSpitzer, @snazy

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java [hunk 1]
```diff
@@ -84,6 +85,12 @@ private TableMetadata loadTableMetadata(String metadataLocation, Reference refer
         NessieUtil.tableMetadataFromIcebergTable(io(), table, metadataLocation);
     Map<String, String> newProperties = Maps.newHashMap(deserialized.properties());
     newProperties.put(NESSIE_COMMIT_ID_PROPERTY, reference.getHash());
+    // To prevent accidental deletion of files that are still referenced by other branches/tags,
+    // setting GC_ENABLED to false. So that all Iceberg's gc operations like expire_snapshots,
+    // remove_orphan_files, drop_table with purge will fail with an error.
+    // Nessie CLI will provide a reference aware GC functionality for the expired/unreferenced
+    // files.
+    newProperties.put(TableProperties.GC_ENABLED, "false");
```
Comment (Fokko) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:93]: This way we disable the GC'ing of the tables completely. Are there cases where we still want Iceberg to do the garbage collection?

Comment (Fokko) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java:93]: That makes sense, thanks!

**Comment by Fokko** [comment]
Thanks @ajantha-bhat 🙌🏻

**Comment by ajantha-bhat** [comment]
@nastra: Let us discuss the scope of 1.0.0 in today's sync. I believe it can accommodate few issues as well like this one and https://github.com/apache/iceberg/pull/5754.

**Pull request merged**